ci/check-documentation.py

```diff
@@ -39,12 +39,31 @@ def find_md_files():
 def main():
     all_passed = True
 
+    md_check_cases = (OMZ_ROOT / 'models' / 'intel' / 'index.md',
+                     OMZ_ROOT / 'models' / 'public' / 'index.md',
+                     OMZ_ROOT / 'demos' / 'README.md')
+
+    all_md_files = tuple(find_md_files())
+
     def complain(message):
         nonlocal all_passed
         all_passed = False
         print(message, file=sys.stderr)
 
+    md_dict = dict()
+    for check_case_path in md_check_cases:
+        valid_md_links = set()
+        for md_file in all_md_files:
+            md_file_parents = tuple(md_file.parents)[1:]
+            if check_case_path.parent in md_file_parents:
+                md_index = md_file_parents.index(check_case_path.parent)
+                if not any((parent_dir / 'README.md').exists() for parent_dir in md_file_parents[:md_index]):
+                    valid_md_links.add(md_file)
```

Comment:
Using a `set` here is unnecessary - all elements in `all_md_files` are different, so there's no way you could add duplicate elements.